File: leetcode/1953-maximum-number-of-weeks-for-which-you-can-work/numberOfWeeks.ts
function numberOfWeeks(milestones: number[]): number {
  let restWeeks = 0;
  let longestWeeks = 0;

  for (const milestone of milestones) {
    longestWeeks = Math.max(milestone, longestWeeks);
    restWeeks = restWeeks + milestone;
  }
  restWeeks = restWeeks - longestWeeks;

  if (longestWeeks > restWeeks + 1) {
    longestWeeks = restWeeks + 1;
  }

  return restWeeks + longestWeeks;
}

export default numberOfWeeks;

File: leetcode/1953-maximum-number-of-weeks-for-which-you-can-work/numberOfWeeks.rs
use crate::Solution;

impl Solution {
    pub fn number_of_weeks(milestones: Vec<i32>) -> i64 {
        let mut rest_weeks: i64 = 0;
        let mut longest_weeks: i64 = 0;

        for milestone in milestones {
            longest_weeks = longest_weeks.max(milestone as i64);
            rest_weeks = rest_weeks + milestone as i64;
        }

        rest_weeks = rest_weeks - longest_weeks;

        if longest_weeks > rest_weeks + 1 {
            longest_weeks = rest_weeks + 1;
        }

        return longest_weeks + rest_weeks;
    }
}
